src/offset.js

```diff
@@ -82,27 +82,28 @@ jQuery.fn.extend({
 				});
 		}
 
-		var docElem, win,
+		var docElem, win, rect,
 			elem = this[ 0 ],
-			box = { top: 0, left: 0 },
 			doc = elem && elem.ownerDocument;
 
 		if ( !doc ) {
 			return;
 		}
 
-		docElem = doc.documentElement;
+		rect = elem.getBoundingClientRect();
 
-		// Make sure it's not a disconnected DOM node
-		if ( !jQuery.contains( docElem, elem ) ) {
-			return box;
+		// Make sure element is not hidden (display: none) or disconnected
+		if ( !rect.width && !rect.height && !rect.left &&
+			!rect.top && !rect.right && !rect.bottom ) {
```

Comment:
I agree with you, which is why I'm advocating a test like `!rect.width && !rect.height` that will not trip up on single-dimension elements such as your no-content div. But at the same time, we must recognize that dimensionless (e.g., no-content _inline_) can be disconnected/display:none false positives (or vice versa)... I don't think the extra top/left checks are worth it, given that even they are imperfect.